geps/gep-3779/index.md

```diff
@@ -246,10 +246,261 @@ Cilium has the concept of CiliumIdentity. Pods are assigned identities derived f
+    //
+    // +optional
+    Attributes []AuthorizationTCPAttributes `json:"attributes,omitempty"`
+}
+
+
+// AuthorizationSource specifies the source of a request.
+// Only one of its fields may be set.
+// +kubebuilder:validation:XValidation:rule="(size(self.identities) > 0 ? 1 : 0) + (size(self.serviceAccounts) > 0 ? 1 : 0) + (size(self.namespaces) > 0 ? 1 : 0) == 1",message="Exactly one of identities, serviceAccounts, or namespaces must be specified."
+type AuthorizationSource struct {
+
+    // Identities specifies a list of identities that are matched by this rule.
+    // A request's identity must be present in this list to match the rule.
+    //
+    // Identities must be specified as SPIFFE-formatted URIs following the pattern:
+    //   spiffe://<trust_domain>/<workload-identifier>
+    //
+    // While the exact workload identifier structure is implementation-specific,
+    // implementations are encouraged to follow the convention of
+    // `spiffe://<trust_domain>/ns/<namespace>/sa/<serviceaccount>`
+    // when representing Kubernetes workload identities.
+    //
+    // Identities for authorization can be derived in various ways by the underlying
+    // implementation. Common methods include:
+    // - From peer mTLS certificates: The identity is extracted from the client's
+    //   mTLS certificate presented during connection establishment.
+    // - From IP-to-identity mappings: The implementation might maintain a dynamic
+    //   mapping between source IP addresses (pod IPs) and their associated
+    //   identities (e.g., Service Account, SPIFFE IDs).
+    // - From JWTs or other request-level authentication tokens. 
+    // 
+    // Note for reviewers: While this GEP primarily focuses on identity-based 
+    // authorization where identity is often established at the transport layer,
+    // some implementations might derive identity from authenticated tokens or sources
+    // within the request itself.
+    //
+    // +optional
+    Identities []string `json:"identities,omitempty"`
+
+    // ServiceAccounts specifies a list of Kubernetes Service Accounts that are
+    // matched by this rule. Each service account must be specified in the format
+    // "<namespace>/<service-account-name>". A request originating from a pod
+    // associated with one of these service accounts will match the rule.
+    //
+    // The ServiceAccounts listed here are expected to exist within the same
+    // trust domain as the targeted workload, which in many environments means
+    // the same Kubernetes cluster. Cross-cluster or cross-trust-domain should
+    // be expressed with Identities field.
+    // +optional
+    ServiceAccounts []string `json:"serviceAccounts,omitempty"`
+
+    // Namespaces specifies a list of Kubernetes Namespaces that are matched
+    // by this rule. A request originating from any pod within one of these
+    // namespaces will match the rule, regardless of its specific Service Account.
```

Comment:
So this is basically just another way of saying "match all ServiceAccounts in these namespaces"? Given the proposed format of ServiceAccounts is `ns/name` above, would `ns/*` be a reasonable alternative way to express this instead of having a separate field in the API?

Reply:
applied the recommendation - ptal